klient/k8s/yaml/yaml.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package yaml
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"sigs.k8s.io/e2e-framework/klient"
+	"sigs.k8s.io/e2e-framework/klient/k8s/resources"
+)
+
+// LoadUnstructuredYAML loads a single-document YAML or JSON file
```

Comment:
As far as the `single-document` term goes:

It is possible to represent multiple yaml documents in a single file. I did not start with supporting that, but should that be reconsidered?

Example:

```
# testdata/two-configmaps.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: example-1
data:
  foo.cfg: |
    foo: bar
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: example-2
data:
  foo.cfg: |
    foo: bar
```

It is a little more complicated to parse out, but not impossible.

The designs _may_ need to be adjust to:
- Have a secondary function that works with a `k8s.ObjectList` or returns a set of `[]*unstructured.Unstructured`